std/zip.d

```diff
@@ -1,61 +1,149 @@
 // Written in the D programming language.
 
 /**
- * Read/write data in the $(LINK2 https://en.wikipedia.org/wiki/Zip_%28file_format%29, zip archive) format.
- * Makes use of the etc.c.zlib compression library.
- *
- * Limitations:
- *      $(UL
- *      $(LI Multi-disk zips not supported.)
- *      $(LI Only Zip version 20 formats are supported.)
- *      $(LI Only supports compression modes 0 (no compression) and 8 (deflate).)
- *      $(LI Does not support encryption.)
- *      )
- *
- * Example:
- * ---
-// Read existing zip file.
-import std.digest.crc, std.file, std.stdio, std.zip;
-
-void main(string[] args)
+Read and write data in the $(LINK2 https://en.wikipedia.org/wiki/Zip_%28file_format%29, zip archive) format.
+
+Standards:
+
+The current implementation mostly conforms to $(LINK2 https://www.iso.org/standard/60101.html, ISO/IEC 21320-1:2015),
+which means,
+$(UL
+$(LI that files can only be stored uncompressed or using the deflate mechanism,)
+$(LI that encryption features are not used,)
+$(LI that digital signature features are not used,)
+$(LI that patched data features are not used, and)
+$(LI that archives may not span multiple volumes.)
+)
+
+Additionally, archives are checked for malware attacks and rejected if detected. This includes
+$(UL
+$(LI $(LINK2 https://news.ycombinator.com/item?id=20352439, zip bombs) which generate gigantic amounts of
+     unpacked data)
+$(LI zip archives that contain overlapping records)
+$(LI chameleon zip archives which generate different unpacked data, depending on the implementation of the
+     unpack algorithm)
+)
+
+The current implementation makes use of the etc.c.zlib compression library.
```

Comment:
I'm not sure, if this is still correct. The implementation uses std.zlib, which probably is based on etc.c.zlib. I'm not sure if this comment is of any help, but it was in the original documentation and so I decided to preserve it.

Reply:
Neither `etc.c.zlib` nor `std.zlib` are a compression library, they merely wrap one. Probably it would be more correct if this was changed to either say that it uses the zlib library or the `etc.c.zlib` bindings / `std.zlib` wrapper for the zlib library.